static/js/components/widgets/ResourcePickerDialog.test.tsx

```diff
@@ -57,121 +58,95 @@ describe("ResourcePickerDialog", () => {
     expect(Object.values(TabIds).length).toBe(uniqueTabIds.size)
   })
 
-  it.each([
-    { mode: RESOURCE_LINK, dropdownExists: true, should: "should" },
-    { mode: RESOURCE_EMBED, dropdownExists: false, should: "should not" }
-  ])(
-    'when in mode "$mode", $should show the "more" dropdown',
-    async ({ mode, dropdownExists }) => {
-      const { wrapper } = await render({ mode })
-      const dropdown = wrapper.find(Dropdown)
-      expect(dropdown.exists()).toBe(dropdownExists)
-    }
-  )
-
   it("should pass some basic props down to the dialog", async () => {
     const { wrapper } = await render()
     const dialog = wrapper.find("Dialog")
     expect(dialog.prop("open")).toBeTruthy()
     expect(dialog.prop("wrapClassName")).toBe("resource-picker-dialog")
   })
 
-  it("should allow focusing and linking a resource, then close the dialog", async () => {
-    const { wrapper } = await render({
-      mode: RESOURCE_LINK
-    })
-    // callback should be 'undefined' before resource is focused
-    expect(wrapper.find("Dialog").prop("onAccept")).toBeUndefined()
-    focusResource(wrapper, resource)
-
-    expect(wrapper.find("Dialog").prop("acceptText")).toBe("Add link")
-
-    act(() => {
-      // @ts-ignore
-      wrapper.find("Dialog").prop("onAccept")()
-    })
-
-    wrapper.update()
-
-    expect(insertEmbedStub.args[0]).toStrictEqual([
-      resource.text_id,
-      resource.title,
-      RESOURCE_LINK
-    ])
-    expect(closeDialogStub.callCount).toBe(1)
-  })
-
-  it("should focusing and embedding a resource", async () => {
-    const { wrapper } = await render({
-      mode: RESOURCE_EMBED
-    })
-    // callback should be 'undefined' before resource is focused
-    expect(wrapper.find("Dialog").prop("onAccept")).toBeUndefined()
-    focusResource(wrapper, resource)
-
-    expect(wrapper.find("Dialog").prop("acceptText")).toBe("Embed resource")
+  it.each([
+    { mode: RESOURCE_LINK, attaching: "linking", acceptText: "Add link" },
+    {
+      mode:       RESOURCE_EMBED,
+      attaching:  "embedding",
+      acceptText: "Embed resource"
+    }
+  ])(
+    "should allow focusing and $attaching a resource, then close the dialog",
+    async ({ mode, acceptText }) => {
+      const { wrapper } = await render({ mode })
+      // callback should be 'undefined' before resource is focused
+      expect(wrapper.find(Dialog).prop("onAccept")).toBeUndefined()
+      focusResource(wrapper, resource)
 
-    act(() => {
-      // @ts-ignore
-      wrapper.find("Dialog").prop("onAccept")()
-    })
+      expect(wrapper.find(Dialog).prop("acceptText")).toBe(acceptText)
 
-    wrapper.update()
+      act(() => {
+        wrapper.find(Dialog).prop("onAccept")?.()
+      })
 
-    expect(insertEmbedStub.args[0]).toStrictEqual([
-      resource.text_id,
-      resource.title,
-      RESOURCE_EMBED
-    ])
-  })
+      wrapper.update()
 
-  it("should pass basic props to ResourcePickerListing", async () => {
```

Comment:
I don't think this test added anything not covered in `"passes the correct props to ResourcePickerListing when main tab $index is clicked"`, so I just removed this test.